keyservice/keyservice.go

```diff
@@ -82,3 +82,22 @@ func KeyFromMasterKey(mk keys.MasterKey) Key {
 		panic(fmt.Sprintf("Tried to convert unknown MasterKey type %T to keyservice.Key", mk))
 	}
 }
+
+func MasterKeyTypeToString(mk keys.MasterKey) string {
+	switch mk := mk.(type) {
+	case *pgp.MasterKey:
+		return "pgp"
+	case *gcpkms.MasterKey:
+		return "gcp_kms"
+	case *hcvault.MasterKey:
+		return "hc_vault"
+	case *kms.MasterKey:
+		return "kms"
+	case *azkv.MasterKey:
+		return "azure_kv"
+	case *age.MasterKey:
+		return "age"
+	default:
+		panic(fmt.Sprintf("Tried to convert unknown MasterKey type %T to string", mk))
+	}
+}
```

Comment:
I would suggest to add a new field or a method to the `MasterKey` struct instead to keep it closer to the source. When new key types are added this might be missed.

Reply:
Good idea. Method seems to fit nicely.